### PR TITLE
Support pandas v2

### PR DESCRIPTION
**Pull request by corneliusroemer**
Support pandas v2.

Pathogen CI uses whatever pandas version is installed by conda base if augur doesn't have conflicting requirements, which is v1 right now, so I made an ephemeral test PR to exclude v2 to test pathogen CI with v2 and it passed, see #1672

## Related issue(s)

Resolves #1471
Part of #1473 

## Checklist

- [x] Automated checks pass
- [x] [Check][1] if you need to add a changelog message
- [x] [Check][2] if you need to add tests
- [x] [Check][3] if you need to update docs

[1]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#updating-the-changelog
[2]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#testing
[3]: https://github.com/nextstrain/augur/blob/-/docs/contribute/DEV_DOCS.md#when-to-update

### CODE REVIEW

**Review by genehack — APPROVED** [review]

**Review by victorlin — APPROVED** [review]

**Comment by victorlin** [comment]
I've moved the post-merge checklist to the task list in https://github.com/nextstrain/augur/issues/1473

**Comment by corneliusroemer** [comment]
Not merging this yet so we can release what's on master right now (quite a bit), so we can detect pandas changes in a following release.

File: CHANGES.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### Features
 
+* Support pandas v2. [#1671] (@corneliusroemer and @victorlin)
```
Comment (joverlee521) [inline CHANGES.md:7]: Reminder to rebase and update the Changelog so it doesn't get out-of-sync due to the latest release of v26.1.0.

Comment (joverlee521) [reply CHANGES.md:7]: Or merge via CLI to fix it during the merge as explained in https://github.com/nextstrain/augur/issues/1620#issuecomment-2327972371

**Pull request merged**